Chapter 5 Author review (#50)
tidy-survey-r/tidy-survey-book: 05-descriptive-analysis.Rmd

Author: - Remove univariate/bivariate headers and move all measures up one level.
- Move survey_count/survey_tally to be close to each other
- Informative headers rather than function names
- Variance types: need standalone description and definitions: add this after a restructuring

The functions in {srvyr} also play nicely with other tidyverse functions. If we wanted to create our survey object by selecting specific columns, we could use `dplyr::select()`'s special selection functions (such as `starts_with()`, `one_of()`, etc.).

```{r}

> dstrata <- dstrata %>% 

Author: ```suggestion
dstrata <- apistrat %>% 
```

* The `survey_total()` function calculates totals
* The `survey_count()` and `survey_tally()` functions calculate weighted observations by group

> * The `survey_prop()` function calculates proportions 

Author: Isn't survey_prop really a mean so also a measure of centrality?

* Examples: Average 2000 Academic Performance Index in California, the median house price in Canada

> The `survey_mean()` and `survey_median()` functions from {srvyr} calculate mean and median using survey data, respectively. These functions must be called from within `summarize()`.  

Author: All except survey_count and survey_tally must be within summarize. Let's make that point clearly and before getting into these groupings

### Setup

> Recall from [Chapter 05](#c05) the general process for estimation with the {srvyr} package:  

Author: Update reference. Is this the first time we are seeing it, if so, no ref needed

* `.drop`: whether to drop empty groups
* `vartype`: type(s) of variation estimate to calculate, defaults to `se` (standard error)

The steps to use `survey_count()` are: 

* Specify the sample design, 
* Filter subsets using `dplyr::filter()`, if needed, 
* Run `survey_count()`, specifying the required arguments within the function 

> Let's see the weighted count of responses in RECS: 

Reviewer: ```suggestion
If we do not specify any variables in `survey_count()`, the function will output the estimated population count (n) and standard error (n_se).  For example, in the RECS data we can obtain the estimated number of households in the U.S. (the target population) by running the following code: 
```

```{r}
recs_des %>% #  Specify the sample design
  survey_count() # Run `survey_count()`
```

> There are 118,208,250 observations in the survey. <!--TODO: Calculate this-->    

Reviewer: ```suggestion
```

> `srvyr::count()` can take one or many variables. To calculate the number of observations for a combination of variables, such as Region and Division, we run the below:    

Reviewer: ```suggestion
To calculate the estimated number of observations for subgroups, such as Region and Division, we can add the variables of interest into the function.     
```

  survey_count() # Run `survey_count()`
```

There are 118,208,250 observations in the survey. <!--TODO: Calculate this-->    

> `srvyr::count()` can take one or many variables. To calculate the number of observations for a combination of variables, such as Region and Division, we run the below:    

Author: ```suggestion
`survey_count()` can take one or many variables. To calculate the number of observations for a combination of variables, such as Region and Division, we run the below:    
```

```{r}
recs_des %>%
  # Specify the required arguments within the function
  survey_count(Region, Division, name = "N") 
```

> When we run the crosstab, we see 5,638,844 observations from the Northeast's New England division, 15,377,694 from the Northeast's Middle Atlantic division, and so on. <!--TODO: Calculate this-->   

Author: These aren't observations but estimated number of housing units

> When we run the crosstab, we see 5,638,844 observations from the Northeast's New England division, 15,377,694 from the Northeast's Middle Atlantic division, and so on. <!--TODO: Calculate this-->   

Author: ```suggestion
When we run the crosstab, we see 5,638,844 weighted observations from the Northeast's New England division, 15,377,694 from the Northeast's Middle Atlantic division, and so on. <!--TODO: Calculate this-->   
```

Reviewer: I would call this the estimated number of households from the Northeast's New....

* `vartype`: type(s) of variation estimate to calculate, defaults to `se` (standard error)
* `level`: a number or a vector indicating the confidence level, defaults to 0.95
* `deff`: a logical value stating whether the design effect should be returned, defaults to FALSE
* `df`: (for `vartype = 'ci'`), a numeric value indicating degrees of freedom for the t-distribution
  * For the {srvyr} package, this defaults to `NULL`, whereas the {survey} package defaults to `Inf`

> The steps to use `survey_total()` are:

Reviewer: As before, let's drop this.

```{r}
recs_des %>%
  summarize(survey_total(),  
            .groups = "drop")  
```

Note that the result from `recs_des %>% summarize(survey_total(), .groups = "drop")` is equivalent to the `survey_count()` call. However, the `survey_total()` function is called within `summarize`, where as `survey_count()`, like `dplyr::count()`, is not. 

> The difference between `survey_total()` and `survey_count()` is more evident when specifying numeric variables to sum. Let's compute the total cost of electricity in whole dollars from variable `DOLLAREL`. We also calculate an unweighted estimate using `unweighted`. 

Reviewer: ```suggestion
The difference between `survey_total()` and `survey_count()` is more evident when specifying continuous variables to sum. Let's compute the total cost of electricity in whole dollars from variable `DOLLAREL`. We also calculate an unweighted estimate using `unweighted`. 
```

> The difference between `survey_total()` and `survey_count()` is more evident when specifying numeric variables to sum. Let's compute the total cost of electricity in whole dollars from variable `DOLLAREL`. We also calculate an unweighted estimate using `unweighted`. 

Reviewer: We probably need to give some examples as to why we would want to include the unweighted estimate as well.  What does this provide us in analysis?

```{r}
recs_des %>%

>   summarize(survey_total(),  
>             .groups = "drop")  
> ```
> 
> Note that the result from `recs_des %>% summarize(survey_total(), .groups = "drop")` is equivalent to the `survey_count()` call. However, the `survey_total()` function is called within `summarize`, where as `survey_count()`, like `dplyr::count()`, is not.

Author: No groups are used so not sure why we have .groups="drop". Move discussion elsewhere to when group is used.
```suggestion
  summarize(survey_total())  
```

Note that the result from `recs_des %>% summarize(survey_total(), .groups = "drop")` is equivalent to the `survey_count()` call. However, the `survey_total()` function is called within `summarize`, where as `survey_count()`, like `dplyr::count()`, is not.
```

recs_des %>%
  summarize(
    elec_bill = survey_total(DOLLAREL),

>     elec_unweight = unweighted(sum(!is.na(DOLLAREL))),  
>     .groups = "drop"    

Author: ```suggestion
    elec_unweight = unweighted(sum(DOLLAREL))
```

  )
```

> $162,495,237,023 was spent on electricity. The unweighted sum is $5686. <!--TODO: add calculation-->   

Author: This total is the national aggregate estimate. Fixed unweighted sum. Let's add note that unweighted sum is the total electricity expenditure among the *respondents*

Reviewer: What's the timeframe for this?  Is it the amount spent in a year?  Would be helpful for some context.  Something to think about across all the sections as well.

Since we are using the {srvyr} package, we can use `group_by()` to calculate the cost of electricity by different groups. Let's see how much the cost of electricity in whole dollars differed between regions:

```{r}
recs_des %>%
  group_by(Region) %>%

>   cascade(elec_bill = survey_total(DOLLAREL))   

Author: Let's use summarize and not cascade here. Cascade is introduced later.

* Specify the cross tab in `group_by()`,
* Run `survey_mean()` or `survey_prop()` within `summarize()`

> We can calculate the weighted average cost of electricity for each region in the U.S.: 

Reviewer: Let's first do the overall then do the region.  We do overall for count/total, so let's do it here too to help walk respondents through each possibility.

```{r}
recs_des %>%
  group_by(Region) %>%

>   summarize(survey_prop(),   

Author: ```suggestion
  summarize(p=survey_prop(),   
```

> ```{r}

Reviewer: Noting that the warning message goes off the pdf page.

Calculating proportions with design effects is important when analyzing survey data that has been collected using a complex sampling design. Since a complex sampling design means that the probability of selecting each individual in the sample is different, this can lead to the observations in the sample being correlated with each other.

> Design effects account for this correlation and adjust the standard errors of estimates, which can affect the significance of test statistics and the accuracy of confidence intervals. Calculating proportions with design effects is important because it helps to produce accurate estimates of population proportions that account for the complex nature of the survey data. 

Reviewer: I think we need to provide more information around deff.  What do they tell us?  How can we use that in analysis and reporting?  I remember having these same questions when I was learning about them in class so I'm not much help here to flush this out 😄
